examples/todomvc/src/lib.rs

```diff
@@ -65,22 +68,26 @@ impl Component for Model {
             value: "".into(),
             edit_value: "".into(),
         };
+        let focus_ref = NodeRef::default();
         Model {
             link,
             storage,
             state,
+            focus_ref,
         }
     }
 
     fn update(&mut self, msg: Self::Message) -> ShouldRender {
         match msg {
             Msg::Add => {
-                let entry = Entry {
-                    description: self.state.value.clone(),
-                    completed: false,
-                    editing: false,
-                };
-                self.state.entries.push(entry);
+                if self.state.value.trim() != "" {
+                    let entry = Entry {
+                        description: self.state.value.trim().to_string(),
```

Comment:
calling `trim` twice
```suggestion
                let description = self.state.value.trim();
                if !description.is_empty() {
                    let entry = Entry {
                        description: description.to_string(),
```